### PR TITLE
Various fixes and small improvements

### PR DESCRIPTION
**Pull request by dreamer**
This review does not have any leading motif, rather a collection of different small fixes (these are commits, that I had created at some point and they started gathering dust on my local branches).

I recommend reading patches one-by-one.

b1886a94ea093162bb81d34c101d716179d1dd53 is needed for future release builds (so we can include the output of git-describe in version string - this way it'll be easy to correlate build with exact commit)

### CODE REVIEW

File: src/ints/int10_vesa.cpp [hunk 1]
```diff
@@ -576,10 +576,6 @@ void INT10_SetupVESA(void) {
 	for (i=0;i<len;i++) {
 		phys_writeb(0xc0000+int10.rom.used++,string_oem[i]);
 	}
-	switch (svgaCard) {
-	case SVGA_S3Trio:
-		break;
-	}
```
Comment (krcroft) [inline src/ints/int10_vesa.cpp]: Very odd; I suppose the compile has been eliding this away for years!

File: README.md [hunk 2]
```diff
@@ -33,19 +33,19 @@ Read more at
...
 | **Opus**       | Yes (libopus)              | No
-| **OGG/Vorbis** | Yes (built-in)             | Yes - SDL\_sound 1.2 (libvorbis)<sup>[6],‡</sup>
-| **MP3**        | Yes (built-in)             | Yes - SDL\_sound 1.2 (libmpg123)<sup>[6],‡,§</sup>
+| **OGG/Vorbis** | Yes (built-in)             | Yes - SDL\_sound 1.2 (libvorbis)<sup>[6],＊</sup>
+| **MP3**        | Yes (built-in)             | Yes - SDL\_sound 1.2 (libmpg123)<sup>[6],＊,§</sup>
 | **FLAC**       | Yes (built-in)             | No<sup>§</sup>
-| **WAV**        | Yes (built-in)             | Yes - SDL\_sound 1.2 (internal)<sup>[7],‡</sup>
-| **AIFF**       | No                         | Yes - SDL\_sound 1.2 (internal)<sup>[7],‡</sup>
+| **WAV**        | Yes (built-in)             | Yes - SDL\_sound 1.2 (internal)<sup>[7],＊</sup>
+| **AIFF**       | No                         | Yes - SDL\_sound 1.2 (internal)<sup>[7],＊</sup>
```
Comment (krcroft) [inline README.md]: Didn't know those existed; very nice!

File: src/hardware/memory.cpp [hunk 3]
```diff
@@ -472,17 +472,21 @@ bool mem_unalignedreadd_checked(PhysPt address, Bit32u * val) {
 	return false;
 }
 
-bool mem_unalignedwritew_checked(PhysPt address,Bit16u val) {
-	if (mem_writeb_checked(address,(Bit8u)(val & 0xff))) return true;val>>=8;
-	if (mem_writeb_checked(address+1,(Bit8u)(val & 0xff))) return true;
```
Comment (krcroft) [inline src/hardware/memory.cpp]: The warning is right; I skimmed right past the val>>=8 over there on the right.
Logic and readability is normal now.

File: .github/workflows/linux.yml [hunk 4]
```diff
@@ -1,70 +1,40 @@
...
+            flags: -c gcc
+            max_warnings: 316
+          - name: GCC-9 (Ubuntu 18.04)
+            os: ubuntu-18.04
+            flags: -c gcc -v 9
+            max_warnings: 339
+          - name: Clang-8 (Ubuntu 18.04)
+            os: ubuntu-18.04
+            flags: -c clang -v 8
+            max_warnings: 175
```
Comment (krcroft) [inline .github/workflows/linux.yml]: Keeps getting leaner and even more readable.
:+1:

File: .github/workflows/windows.yml [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 on: push
 env:
-  MAX_WARNINGS_GCC_32bit_Debug:     243
-  MAX_WARNINGS_GCC_64bit_Debug:     359
-  MAX_WARNINGS_Clang_32bit_Debug:   105
-  MAX_WARNINGS_Clang_64bit_Debug:   160
+  MAX_WARNINGS_GCC_32bit_Debug:     234
+  MAX_WARNINGS_GCC_64bit_Debug:     350
+  MAX_WARNINGS_Clang_32bit_Debug:   100
+  MAX_WARNINGS_Clang_64bit_Debug:   155
```
Comment (krcroft) [inline .github/workflows/windows.yml]: .. and I noticed the Coverity job checked in with 2 fewer issues too! Nice work @dreamer .

File: src/ints/int10_vesa.cpp (hunk 1 above)
Comment (dreamer) [reply src/ints/int10_vesa.cpp]: Yeah, this is weird - as if someone planned to implement something, or left refactoring halfway done… (this generated 4 warnings IIRC).

File: README.md (hunk 2 above)
Comment (dreamer) [reply README.md]: The wiki article about "Asterisk" lists lots and lots of them :)

File: src/hardware/memory.cpp (hunk 3 above)
Comment (dreamer) [reply src/hardware/memory.cpp]: Yeah, me too. And lack of whitespace made it easy to miss 'return' in there as well.

**Comment by dreamer** [comment]
Thanks! I can do a cleanup in my local branches now :)

File: src/ints/int10_vesa.cpp (hunk 1 above)
Comment (krcroft) [reply src/ints/int10_vesa.cpp]: That's like breaking the sound barrier with code.. achieved more warnings than lines of code.

**Pull request merged**